camel/messages/base.py

```diff
@@ -282,6 +288,81 @@ def to_openai_user_message(self) -> OpenAIUserMessage:
                     },
                 ],
             }
+        elif self.video_path is not None:
+            if not os.path.exists(self.video_path):
+                raise FileNotFoundError(
+                    f"Video file {self.video_path} does not exist."
+                )
+
+            video_format: str = self.video_path.split(".")[-1].lower()
+            if video_format not in OpenAIVideoType:
+                raise ValueError(
+                    f"Video type {video_format} "
+                    f"is not supported by OpenAI vision model"
+                )
+
+            base64Frames = []
+            video = cv2.VideoCapture(self.video_path)
+            while video.isOpened():
+                success, frame = video.read()
+                if not success:
+                    break
+
+                # Get the dimensions of the frame
+                height, width = frame.shape[:2]
+
+                # Calculate the new dimensions while maintaining the aspect ratio
+                new_width = Constants.DEFAULT_IMAGE_SIZE
+                aspect_ratio = width / height
+                new_height = int(new_width / aspect_ratio)
+
+                # Resize the frame
+                resized_frame = cv2.resize(frame, (new_width, new_height))
+
+                # Encode the frame as JPEG
+                success, buffer = cv2.imencode(  # type: ignore[assignment]
+                    f".{OpenAIImageType.JPEG}", resized_frame
+                )
+
+                # Convert the encoded frame to base64
+                base64Frame = base64.b64encode(buffer).decode("utf-8")  # type: ignore[arg-type]
+
+                # Append the base64 encoded frame to the list
+                base64Frames.append(base64Frame)
+
+            # Release the video capture object
+            video.release()
+
+            # Define a list to hold extracted images from base64Frames
+            # Extract images at intervals specified by Constants.IMAGE_EXTRACTION_INTERVAL
+            # Using list comprehension to select an image at each interval
+            image_extraction_list = [
+                x
+                for x in base64Frames[0 :: Constants.IMAGE_EXTRACTION_INTERVAL]
+            ]
+
+            video_content: List[Any] = []
+            video_content.append(
+                {
+                    "type": "text",
+                    "text": "These are frames from a video that I want to upload. Generate a compelling description that I can upload along with the video.",
+                }
+            )
+            for image in image_extraction_list:
+                item = {
+                    "type": "image_url",
+                    "image_url": {
+                        "url": f"data:image/jpg;base64,{image}",
+                    },
+                }
+                video_content.append(item)
+            return {
+                "role": "user",
+                "content": video_content,
+            }
```

Comment:
seems we lost the information provided from `content` and `image`, we also need to append that information